components/script/dom/gpudevice.rs

```diff
@@ -63,6 +75,74 @@ impl GPUDevice {
     }
 }
 
+impl GPUDevice {
+    fn resolve_create_buffer_mapped(
```

Comment:
This should be marked unsafe, since it accepts an arbitrary integer, converts it to a pointer, and dereferences it by using it to create a slice.